ppml/trusted-deep-learning/base/load_save_encryption_ex.py

```diff
@@ -0,0 +1,220 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from torch.utils.data import DataLoader
+import torch
+import torch.optim as optim
+from contextlib import nullcontext
+import torch.nn.functional as F
+import os
+import torch.nn as nn
+import time
+from bigdl.nano.pytorch.patching import patch_encryption
+import logging
+from bigdl.ppml.kms.client import generate_primary_key, generate_data_key, get_data_key_plaintext
+from datasets.load import load_from_disk
+import argparse
+from transformers import BertTokenizer, BertModel, AdamW
+
+parser = argparse.ArgumentParser(description="PyTorch PERT Example")
+parser.add_argument("--local-only", action="store_true", default=False,
+                    help="If set to true, then load model from disk")
+parser.add_argument("--model-path", type=str, default="/ppml/model",
+                    help="Where to load model")
+parser.add_argument("--dataset-path", type=str, default="/ppml/dataset",
+                    help="Where to load original dataset")
+
+
+# python3 load_save_ex.py --local-only --model-path /ppml/model --dataset-path /ppml/save-datasets/train/
+args = parser.parse_args()
+
+
+# Define APPID and APIKEY in os.environment
+APPID = os.environ.get('APPID')
+APIKEY = os.environ.get('APIKEY')
+
+
+encrypted_primary_key_path = ""
+encrypted_data_key_path = ""
+
+EHSM_IP = os.environ.get('ehsm_ip')
+EHSM_PORT = os.environ.get('ehsm_port', "9000")
+
+if args.local_only:
+    checkpoint = args.model_path
+    tokenizer = BertTokenizer.from_pretrained(
+        checkpoint, model_max_length=512, local_files_only=True)
+else:
+    checkpoint = 'hfl/chinese-pert-base'
+    tokenizer = BertTokenizer.from_pretrained(checkpoint, model_max_length=512)
+
+
+# prepare environment
+def prepare_env():
+    """
+    |1. Check whether arguments required by KMS is set.
+    |2. Apply patch to torch.
+    """
+    if APPID is None or APIKEY is None or EHSM_IP is None:
+        print("Please set environment variable APPID, APIKEY, ehsm_ip!")
+        exit(1)
+    generate_primary_key(EHSM_IP, EHSM_PORT)
+    global encrypted_primary_key_path
+    encrypted_primary_key_path = "./encrypted_primary_key"
+    generate_data_key(EHSM_IP, EHSM_PORT, encrypted_primary_key_path, 32)
+    global encrypted_data_key_path
+    encrypted_data_key_path = "./encrypted_data_key"
+    patch_encryption()
+
+
+# Get a key from kms that can be used for encryption/decryption
+def get_key():
+    return get_data_key_plaintext(EHSM_IP, EHSM_PORT, encrypted_primary_key_path, encrypted_data_key_path)
+
+
+def save_encrypted_dataset(dataset_path, save_path, secret_key):
+    dataset = load_from_disk(dataset_path, keep_in_memory=True)
+    # This will save the encrypted dataset into disk
+    torch.save(dataset, save_path, encryption_key=secret_key)
+
+
+class Dataset(torch.utils.data.Dataset):
```

Comment:
can this dataset be reused?

Reply:
Fixed in PR: https://github.com/intel-analytics/BigDL/pull/7197